src/Test/Utilities/Portable/Platform/Custom/OSVersion.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Roslyn.Test.Utilities
+{
+    public static class OSVersion
+    {
+        /// <summary>
+        /// True when the operating system is at least Windows version 8
+        /// </summary>
+        public static bool IsWin8 =>
+#if NET46 || NETCOREAPP2_0
+            System.Environment.OSVersion.Version.Build >= 9200;
+#else
+            throw new InvalidOperationException();
```

Comment:
:bulb: `PlatformNotSupportedException`?

Reply:
We haven't been using that for our `#if` solutions. Generally InvalidOperationException and NotSupportedException. I will look into changing this holistically in the change I'm working on right now. This PR is a series of changes to get our tests running on CoreClr.

Reply:
https://github.com/dotnet/roslyn/projects/16#card-10698974